time/src/test/java/net/jqwik/api/time/DatesTests.java

```diff
@@ -1,20 +1,384 @@
+		void between() {
+			Optional<ExhaustiveGenerator<LocalDate>> optionalGenerator = Dates.dates()
+																			  .between(LocalDate.of(42, Month.DECEMBER, 30), LocalDate
+																																	 .of(43, Month.JANUARY, 2))
+																			  .exhaustive();
+			assertThat(optionalGenerator).isPresent();
+
+			ExhaustiveGenerator<LocalDate> generator = optionalGenerator.get();
+			assertThat(generator.maxCount()).isEqualTo(744); // Cannot know the number of filtered elements in advance
+			assertThat(generator).containsExactly(LocalDate.of(42, Month.DECEMBER, 30), LocalDate.of(42, Month.DECEMBER, 31), LocalDate
+																																	  .of(43, Month.JANUARY, 1), LocalDate
+																																										 .of(43, Month.JANUARY, 2));
+		}
+
+		@Property(tries = 5)
+		void onlyMonthsWithSameYearAndDayOfMonth() {
+			Optional<ExhaustiveGenerator<LocalDate>> optionalGenerator = Dates.dates().yearBetween(1997, 1997).dayOfMonthBetween(17, 17)
+																			  .onlyMonths(Month.MARCH, Month.OCTOBER, Month.DECEMBER)
+																			  .exhaustive();
+			assertThat(optionalGenerator).isPresent();
+
+			ExhaustiveGenerator<LocalDate> generator = optionalGenerator.get();
+			assertThat(generator.maxCount()).isEqualTo(12); // Cannot know the number of filtered elements in advance
+			assertThat(generator).containsExactly(LocalDate.of(1997, Month.MARCH, 17), LocalDate.of(1997, Month.OCTOBER, 17), LocalDate
+																																	  .of(1997, Month.DECEMBER, 17));
+		}
+
+		@Property(tries = 5)
+		void onlyDaysOfWeekWithSameYearAndMonth() {
+			Optional<ExhaustiveGenerator<LocalDate>> optionalGenerator = Dates.dates().yearBetween(2020, 2020).monthBetween(12, 12)
+																			  .onlyDaysOfWeek(DayOfWeek.MONDAY, DayOfWeek.THURSDAY)
+																			  .exhaustive();
+			assertThat(optionalGenerator).isPresent();
+
+			ExhaustiveGenerator<LocalDate> generator = optionalGenerator.get();
+			assertThat(generator.maxCount()).isEqualTo(372); // Cannot know the number of filtered elements in advance
+			assertThat(generator).containsExactly(LocalDate.of(2020, Month.DECEMBER, 3), LocalDate.of(2020, Month.DECEMBER, 7), LocalDate
+																																		.of(2020, Month.DECEMBER, 10), LocalDate
+																																											   .of(2020, Month.DECEMBER, 14), LocalDate
+																																																					  .of(2020, Month.DECEMBER, 17), LocalDate
+																																																															 .of(2020, Month.DECEMBER, 21), LocalDate
+																																																																									.of(2020, Month.DECEMBER, 24), LocalDate
+																																																																																		   .of(2020, Month.DECEMBER, 28), LocalDate
+																																																																																												  .of(2020, Month.DECEMBER, 31));
+		}
+
+	}
+
+	@Group
+	class EdgeCasesTests {
```

Comment:
Can you please provide feedback on the `EdgeCasesTests` classes?

Some of the tests are currently not working.
For example:
```
@Property(tries = 5)
void between() {

	MonthArbitrary months = Dates.months().between(Month.MARCH, Month.AUGUST);
	Set<Month> edgeCases = collectEdgeCases(months.edgeCases());
	assertThat(edgeCases).hasSize(2);
	assertThat(edgeCases).contains(Month.MARCH, Month.AUGUST);

}
```

Reply:
What's the error?

Reply:
```
java.lang.AssertionError:
    Expected size:<2> but was:<0> in:
    <[]>
```

Reply:
I'll have a closer look. 
BTW, `@Property(tries = 5)` doesnt make sense IMO since edge case generation is deterministic.
I'd go with `@Example`

Reply:
If you take away the "smallest" and "largest" month, no edge case remains which is what I'd expect. So I think you're good here.

Reply:
Similiar to edge cases, exhaustive generation is also deterministic and thus I'd go with `@Example` there.

Reply:
`months = months.edgeCases(monthConfig -> monthConfig.add(min).add(max));`

I fixed it with that and I am working on the other edge cases. Should I remove it?

Reply:
As user I expect the min and max values as edgecases

Reply:
Using `EdgeCases.Config.add(..)` has the drawback that values added this way cannot be shrunk. 
Given there are only 12 months anyway, I'd drop it here.

Reply:
> As user I expect the min and max values as edgecases

You could manipulate the original selection to go from min to max. Then those edge cases would get in automatically.

Reply:
> Using `EdgeCases.Config.add(..)` has the drawback that values added this way cannot be shrunk.
> Given there are only 12 months anyway, I'd drop it here.

Sure that this is a problem? The shrinking tests are not bader now.

Reply:
I'll upload it so you can take a closer look.